starknet-replay/src/histogram/mod.rs

```diff
@@ -0,0 +1,281 @@
+//! This module is responsible for rendering and saving the SVG image of the
+//! libfunc histogram.
+//!
+//! It uses the `plotter` library to generate the histogram
+//! and rendering.
+//!
+//! SVG has been chosen because, as a vector graphics format, it makes reading
+//! text easy and zooming doesn't degrade the quality.
+//!
+//! The file `mod.rs` contains the public interface. The main way to use this
+//! module is by calling the function `export` to render and save the SVG image.
+//! The file `plot.rs` contains the call to the `plotters` library.
+
+use std::ops::{Add, Div};
+use std::path::PathBuf;
+
+use super::runner::replay_statistics::ReplayStatistics;
+use crate::error::HistogramError;
+use crate::histogram::plot::render;
+
+mod plot;
+
+/// This alias improves readability of the histogram parameters.
+type Pixel = u32;
+
+/// This struct contains the variable configuration parameters for rendering the
+/// histogram image.
+///
+/// Configuration parameters that don't depend on `ReplayStatistics` are
+/// hardcoded.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Config {
+    /// The width of the SVG image of the histogram in pixels.
+    pub width: Pixel,
+
+    /// The height of the SVG image of the histogram in pixels.
+    pub height: Pixel,
+
+    /// The max number shown on the y axis of the histogram.
+    pub max_y_axis: usize,
+
+    /// Number of pixels used below the x-axis for the labels.
+    pub x_label_area: Pixel,
+}
+impl Config {
+    /// Construct a new `Config` object.
+    ///
+    /// # Arguments
+    ///
+    /// - `libfunc_stats`: the data to be plotted on the histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if:
+    ///
+    /// - There is a math overflow when computing the `Config` parameters
+    /// - There is a truncation when casting from `usize` to `u32`.
+    pub fn new(libfunc_stats: &ReplayStatistics) -> Result<Self, HistogramError> {
+        let max_frequency = libfunc_stats
+            .get_highest_frequency()
+            .ok_or(HistogramError::Empty)?;
+        let number_of_buckets = libfunc_stats.get_number_of_libfuncs();
+        let x_label_area = Self::calc_x_label_area(libfunc_stats)?;
+        let width = Self::calc_width(number_of_buckets)?;
+        let max_y_axis = Self::calc_max_y_axis(max_frequency)?;
+        let height = Self::calc_height(max_y_axis, x_label_area)?;
+
+        tracing::info!("Number of buckets {number_of_buckets}");
+        tracing::info!("Max y axis {max_y_axis}");
+
+        Ok(Config {
+            width,
+            height,
+            max_y_axis,
+            x_label_area,
+        })
+    }
+
+    /// Calculate the space required required to render the x axis labels.
+    ///
+    /// The space is returned in pixels.
```

Comment:
Useless thing in a comment! These docs are always put next to the type signature, so you don't need to describe types; if you do, your types are probably not descriptive enough!